src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -52,15 +52,26 @@ internal sealed class VariableState
         /// Contains a result type which tells us whether the expression may be null,
         /// and an l-value type which tells us whether we can assign null to the expression.
         /// </summary>
+        [DebuggerDisplay("{GetDebuggerDisplay(), nq}")]
         private readonly struct VisitResult
         {
             public readonly TypeWithState RValueType;
             public readonly TypeSymbolWithAnnotations LValueType;
+            public readonly int Slot;
 
-            public VisitResult(TypeWithState resultType, TypeSymbolWithAnnotations lValueType)
+            public VisitResult(TypeWithState rValueType, TypeSymbolWithAnnotations lValueType, int slot)
             {
-                RValueType = resultType;
+                RValueType = rValueType;
                 LValueType = lValueType;
+                Slot = slot;
+            }
+
+            private string GetDebuggerDisplay()
+            {
+                string type = LValueType.HasType ?
```

Comment:
In my testing locally, I added an assert that the `TypeSymbol` of `LValueType` and `RValueType` are always the same and it does not fail, so I don't think this is going to actually do anything. I think it would be better to just put both of them in the display.